### PR TITLE
fix(FileFollower): refactor iterator

### PR DESCRIPTION
**Pull request by aleksbykov**
Poller instance created on each iteration and
file registered  each time. If line is None or
not end on "\n" file is not unregistered and
Poller object could conitnue to eat memory while
will not be removed by garbage collector.

Create 1 Poller object and register/unregister
 file in Poller object only per iteration

Thanks to @dkropachev for help and providing solution

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

File: sdcm/utils/common.py [hunk 1]
```diff
@@ -1598,18 +1598,21 @@ def __init__(self, filename, thread_obj):
...
                 if poller.poll(100):
                     line += input_file.readline()
                 if not line or not line.endswith('\n'):
                     time.sleep(0.1)
                     continue
                 poller.unregister(input_file)
+                registered = False
                 yield line
                 line = ''
-            yield line
```
Comment (soyacz) [inline sdcm/utils/common.py:1612]: won't removing this cause `StopIteration` exception somewhere?

**Review by soyacz — COMMENTED** [review]
How did you find this issue? Was there some visible memory leak or you just found it by accident?

File: sdcm/utils/common.py (hunk 1 above)
Comment (aleksbykov) [reply sdcm/utils/common.py:1612]: it was lost during recreating pr. Return it back. it have to on place

**Comment by aleksbykov** [comment]
> How did you find this issue? Was there some visible memory leak or you just found it by accident?

During investigation the jobs for 5.2 and 2023.1 https://jenkins.scylladb.com/job/enterprise-2023.1/job/longevity/job/longevity-schema-topology-changes-12h-test/4/, where job mostrly failed because hydra eat all memory in 4-5 hours of running. With help of @dkropachev who provide solution, we found that possible place where memory could be leaked. But looks that there is also another place where memory leak could happened.

**Comment by aleksbykov** [comment]
@soyacz @temichus @fruch Could you review?

**Comment by fruch** [comment]
> > How did you find this issue? Was there some visible memory leak or you just found it by accident?
> 
> During investigation the jobs for 5.2 and 2023.1 https://jenkins.scylladb.com/job/enterprise-2023.1/job/longevity/job/longevity-schema-topology-changes-12h-test/4/, where job mostrly failed because hydra eat all memory in 4-5 hours of running. With help of @dkropachev who provide solution, we found that possible place where memory could be leaked. But looks that there is also another place where memory leak could happened.

we commonly have those type of things introduced to SCT,  can you share some of the tools/methods used to identify the origin of those leaks ? so other might be able to investigate such issues ?

**Pull request merged**